### DISCUSSION TITLE
How to reproduce the CtCp section figure from Dolby Whitepaper?

### DISCUSSION THREAD

**Discussion by Zemax27** [colour-science/colour#1246] [Q&A] [answered]
Hello, I'm trying to plot the visible spectrum in CtCp plane

```
plot_visible_spectrum_section(
    model="ICtCp",
    section_colours="RGB",
    contour_colours="RGB",
    section_opacity=1.0,
    )
```

The code above generate the following figure
![image](https://github.com/colour-science/colour/assets/130750380/2e4f2bc7-c177-428d-8f50-d6651db40b7a)

CpCt range limits are supposed to be -0.5, +0.5 
Am I missing something to correctly generate the figure ?

**Comment by thomasmansencal** [comment]
Hello @Zemax27,

The figure is looking correct, here is another variant in 3D using [colour-visuals](https://colour-visuals.readthedocs.io):

![image](https://github.com/colour-science/colour/assets/99779/b83ef1c8-d405-4707-a0a5-1b7058980861)

You can verify some of the range yourself as follows:

```python
print(colour.XYZ_to_ICtCp(colour.wavelength_to_XYZ([400, 500, 600, 700])))
```

```text
[[ 0.011285    0.06543798 -0.00336796]
 [ 0.09560718 -0.05734131 -0.0856163 ]
 [ 0.12840277 -0.07889385  0.11211145]
 [ 0.01395698  0.00170128  0.03504058]]
```

**Comment by Zemax27** [comment]
Hi @KelSolaar,
Thanks for answering my question and congratulation because your work is amazing.
What I’m trying to do is to reproduce the figure 8 of the Dolby white paper about ICtCP
https://professional.dolby.com/siteassets/pdfs/ictcp_dolbywhitepaper_v071.pdf
In the mathematical model, there is a matrix multiplication by a scalar :
<img width="427" alt="image" src="https://github.com/colour-science/colour/assets/130750380/efbbd2b3-e2cd-4f3a-a46e-1b949209f002">
The ITU standard also uses -0.5 to 0.5 as range for CtCp values. So you can understand that I feel a bit odd having a color volume “smaller” than the standard.

Is it your graphical representation limited to 8bits RGB values? Maybe this could explain the difference since RGB to ICtCp considers a max of 12bits code values

**Reply by thomasmansencal** [reply]
Hello,

The figure shows ITU-R BT.2020 volume covering the PQ range, i.e. unit RGB cube scaled by 10000, in ICtCp:

![image](https://github.com/colour-science/colour/assets/99779/8c0a77cc-79f0-450d-a7d8-3b5f08acc80d)

There is no easy way to scale our unit RGB cubes atm, so I will add a note to be able to do that!

Cheers,

Thomas

**Reply by thomasmansencal** [reply]
In the latest `develop` branch:

```python
colour.plotting.CONSTANTS_COLOUR_STYLE.colour.colourspace.cctf_encoding = lambda x: colour.cctf_encoding(x / 1000)

plot_RGB_colourspace_section(
    "ITU-R BT.2020",
    model="ICtCp",
    section_colours="RGB",
    size=10000,
    segments=256
)
colour.plotting.CONSTANTS_COLOUR_STYLE.colour.colourspace.cctf_encoding = colour.cctf_encoding
```

![image](https://github.com/colour-science/colour/assets/99779/f74c8d5a-5aa9-440d-98c6-15d418e52182)


You might need multiple sections to fill the entire top down view because nothing says that 50% represent the maximal extent.

**Reply by Zemax27** [reply, marked as answer]
I'm glad this helped to improve the code features.
I've checked your latest commits and I realized it could be done using the latest release.

```
from colour.models import RGB_COLOURSPACE_BT2020
from colour.utilities import is_trimesh_installed
from colour.geometry import primitive_cube
from colour import RGB_to_XYZ
from colour.plotting import section, CONSTANTS_COLOUR_STYLE
from colour import cctf_encoding


CONSTANTS_COLOUR_STYLE.colour.colourspace = RGB_COLOURSPACE_BT2020
RGB_COLOURSPACE_BT2020.cctf_encoding = lambda x: cctf_encoding(x / 1000)
vertices, faces, _outline = primitive_cube(1, 1, 1, 256, 256, 256)
XYZ_vertices = RGB_to_XYZ(
    (vertices["position"] + 0.5)*10000, RGB_COLOURSPACE_BT2020
)
if is_trimesh_installed:
    from trimesh import Trimesh

    hull = Trimesh(XYZ_vertices, faces, process=False)
    section.plot_hull_section_colours(hull, model='ICtCp', origin=0.7, section_colours="RGB", x_label='Ct', y_label='Cp')
```

![image](https://github.com/colour-science/colour/assets/130750380/1254e5f1-65a5-4264-9319-aaed036ac614)


Thank you very much for the support !

**Reply by thomasmansencal** [reply]
My pleasure!
